src/elliptic/curves/traits.rs

```diff
@@ -15,30 +15,28 @@
 */
 
 use BigInt;
-use Point;
-
-pub trait CurveConstCodec {
-    fn get_base_point() -> Point;
-    fn get_q() -> BigInt;
-}
-
-/// Secret Key Codec: BigInt <> SecretKey
-pub trait SecretKeyCodec<EC> {
-    fn new_random(s: &EC) -> Self;
-    fn from_big_int(s: &EC, n: &BigInt) -> Self;
 
+pub trait ECScalar<SK> {
+    fn new_random() -> Self;
+    fn get_element(&self) -> SK;
+    fn set_element(&mut self, element: SK);
+    fn from_big_int(n: &BigInt) -> Self;
     fn to_big_int(&self) -> BigInt;
+    fn get_q(&self) -> BigInt;
+    fn add(&self, other: &SK) -> Self;
+    fn mul(&self, other: &SK) -> Self;
+    fn sub(&self, other: &SK) -> Self;
 }
 
-/// Public Key Codec: Point <> PublicKey
-pub trait PublicKeyCodec<EC, SK> {
-    const KEY_SIZE: usize;
-    const HEADER_MARKER: usize;
-
+// TODO: add a fn is_point
+pub trait ECPoint<PK, SK> {
+    fn new() -> Self;
```

Comment:
Consider naming it `generator()` (instead of `new()`).

Reply:
done. good idea!!